src/vs/workbench/parts/lib/node/settingsVariables.ts

```diff
@@ -0,0 +1,26 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+'use strict';
+
+import * as Types from 'vs/base/common/types';
+import { AbstractSystemVariables } from 'vs/base/common/parsers';
+import { IConfigurationService } from 'vs/platform/configuration/common/configuration';
+
+export class SettingsVariables extends AbstractSystemVariables {
+	private configuration: any;
+	constructor(private configurationService: IConfigurationService) {
+		super();
+		this.configuration = this.configurationService.getConfiguration();
+	}
+
+	protected resolveString(value: string): string {
+		let regexp = /\$\{settings\.(.*?)\}/g;
+		return value.replace(regexp, (match: string, name: string) => {
+			let config = this.configuration;
+			let newValue = new Function('_', 'try {return _.' + name + ';} catch (ex) { return "";}')(config);
+			return Types.isString(newValue) ? newValue : '';
```

Comment:
@isidorn , I don't think it can be done as a lambda function. The problem is the name of the property is unknown and we could have a hierarchal access such as ${settings.editor.fontFamily}.
The only other way I could think of re-writing this is to use a loop and extract the values one by one.
If you know if any other way or prefer a loop then please do let me know and I'll modify accordingly.